ValidationLibrary/Rules/HasNewestPtcsJenkinsLibRule.cs

```diff
@@ -45,43 +49,76 @@ public async Task Init(IGitHubClient ghClient)
+                     nameof(HasNewestPtcsJenkinsLibRule), RuleName, BranchName);
                 return;
             }
-            var fixedContent = _regex.Replace(jenkinsContent.Content, $"'{LibraryName}@{_expectedVersion}'");
 
-            var branches = await client.Repository.Branch.GetAll(repository.Owner.Login, repository.Name);
-            if (branches.Any(branch => branch.Name == branchName))
+            var reference = await PushFix(client, repository, latest, fixedContent);
+            var pullRequest = new PullRequestRequest
             {
-                _logger.LogInformation("Rule {ruleClass} / {ruleName}, Branch {branchName} already exists, skipping fix.",
-                     nameof(HasNewestPtcsJenkinsLibRule), RuleName, branchName);
+                State = ItemStateFilter.All
+            };
+            var pullRequests = await client.PullRequest.GetAllForRepository(repository.Owner.Login, repository.Name, pullRequest);
+            var openPullRequests = pullRequests.Where(pr => pr.Title == PullRequestTitle && pr.State == ItemState.Open);
+            if (openPullRequests.Any())
+            {
+                _logger.LogInformation("Rule {ruleClass} / {ruleName}, Open pull request already exists. Skipping.", nameof(HasNewestPtcsJenkinsLibRule), RuleName);
+                return;
+            }
+
+            var closed = pullRequests.FirstOrDefault(pr => pr.Title == PullRequestTitle && pr.State == ItemState.Closed && !pr.Merged);
+            if (closed != null && await _gitUtils.PullRequestHasLiveBranch(client, closed))
+            {
+                _logger.LogInformation("Rule {ruleClass} / {ruleName}, Closed pull request with active branch found. Reopening pull request.", nameof(HasNewestPtcsJenkinsLibRule), RuleName);
+                await OpenOldPullRequest(client, repository, closed);
                 return;
             }
-            var branchReference = await client.Git.Reference.CreateBranch(repository.Owner.Login, repository.Name, branchName);
-            var master = await client.Git.Reference.Get(repository.Owner.Login, repository.Name, "heads/master");
 
-            var latest = await client.Git.Commit.Get(repository.Owner.Login, repository.Name, branchReference.Object.Sha);
-            _logger.LogTrace("Latest commit with message {a}", latest.Message);
+            await CreateNewPullRequest(client, repository, reference);
+        }
 
+        /// <summary>
+        /// This takes either the latest commit from master or latest from updated branch if it exists.
+        /// </summary>
+        private async Task<Commit> GetCommitAsBase(IGitHubClient client, Repository repository)
+        {
+            var branches = await client.Repository.Branch.GetAll(repository.Owner.Login, repository.Name);
+            var existingBranch = branches.FirstOrDefault(branch => branch.Name == BranchName);
```

Comment:
Can't use `client.Repository.Branch.Get(repository.Owner.Login, repository.Name, BranchName)`?

Reply:
Good point! I added this to https://github.com/protacon/repository-validator/issues/115